src/examples.test.ts

```diff
@@ -54,7 +55,7 @@ describe("examples tests", () => {
     it(`should correctly convert example ${index}`, () => {
       try {
         // Convert JSON schema to Zod
-        const zodSchema = convertJsonSchemaToZod(example);
+        const zodSchema = convertJsonSchemaToZod(example as unknown as JSONSchema.BaseSchema);
```

Comment:
Why does this not conform to `JSONSchema.BaseSchema` by itself anymore?

Reply:
Typescript is unable to infer the correct type of the imported JSON data, so is defaulting to any. Because of this the type of property $schema are incompatible between the const examples and what the function is expecting.

e.g.
Type `string` is not assignable to type: `"https://json-schema.org/draft/2020-12/schema" | "http://json-schema.org/draft-07/schema#" | "http://json-schema.org/draft-04/schema#" | undefined`

However, I have updated the code to look a little nicer by explicitly enforcing the type of the examples const. It's mostly cosmetic, but it is likely more useful to future developers this way.